includes/template-library/manager.php

```diff
@@ -764,4 +756,21 @@ private function ensure_args( array $required_args, array $specified_args ) {
 
 		return true;
 	}
+
+	private function is_allowed_to_read_template( array $args ): bool {
+		if ( 'remote' === $args['source'] ) {
+			return true;
+		}
+
+		if ( null === $this->wordpress_adapter ) {
+			$this->set_wordpress_adapter( new WordPress_Adapter() );
+		}
+
+		$post_id = intval( $args['template_id'] );
+		$post_status = $this->wordpress_adapter->get_post_status( $post_id );
+		$is_private_or_non_published = ( 'private' === $post_status && ! $this->wordpress_adapter->current_user_can( 'read_private_posts', $post_id ) ) || ( 'publish' !== $post_status );
+
+		return $is_private_or_non_published || ! $this->wordpress_adapter->current_user_can( 'edit_post', $post_id );
```

Comment:
You need to add the methods back into the WP Adapter? `current_user_can` and `get_post_status`